src/devices/honeywell_fan.c

```diff
@@ -0,0 +1,171 @@
+and 'Dim' bits.
+
+    byte[2] >> 3:
+    -------------
+    0x10: Low fan speed
+    0x08: Medium fan speed
+    0x18: High fan speed
+    0x02: Fan off, momentary press of the power button
+    0x05: Delayed light off, extended press of the power button
+    0x1A: Light on/off, momentary press of the 'star' button
+    0x0E: Light dim/brighten, extended press of the 'star' button
+
+*/
+
+static int honeywell_fan_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    int return_code = 0;
+
+    for (int row = 0; row < bitbuffer->num_rows; row++) {
+        int num_bits = bitbuffer->bits_per_row[row];
+
+        if (num_bits != 24) { // Correct number of bits is 24
+            if (decoder->verbose > 1) {
+                fprintf(stderr, "%s: Expected %d bits, got %d.\n", __func__, 21, num_bits); // Max number of bits is 21
+            }
+            continue;
+        }
+
+        uint8_t bytes[3] = {0}; // Max number of bytes is 3
+        bitbuffer_extract_bytes(bitbuffer, row, 0, bytes, 24); // Valid byte offset is 0, Max number of bits is 24
+
+        // Sanity check leading 'fixed' portion
+        if (bytes[0] != 0x16 || bytes[1] != 0xcd) {
+            if (decoder->verbose > 1) {
+                fprintf(stderr, "%s: Expected leading fixed bits 0x16CD, got %x%x.\n", __func__, bytes[0], bytes[1]);
+            }
+            continue;
+        }
+
+        int dimmable = bytes[2] & 0x01;
+        int command = (bytes[2] >> 3) & 0x1f;
+        char command_string[80] = {0};
+
+        switch (command) {
+        case 0x10: // Low fan speed
+            sprintf(command_string, "fan_low");
+            break;
+
+        case 0x08: // Medium fan speed
+            sprintf(command_string, "fan_medium");
+            break;
+
+        case 0x18: // Hign fan speed
+            sprintf(command_string, "fan_high");
+            break;
+
+        case 0x02: // Fan off
+            sprintf(command_string, "fan_off");
+            break;
+
+        case 0x05: // Delayed light off
+            sprintf(command_string, "light_off_delayed");
+            break;
+
+        case 0x1a: // Light on/off
+            sprintf(command_string, "light_on_off");
+            break;
+
+        case 0x0e: // Light dim/brighten
+            sprintf(command_string, "light_dim_brighten");
+            break;
+
+        default:
+            if (decoder->verbose > 1) {
+                fprintf(stderr, "%s: Unknown command: %d\n", __func__, command);
+                continue;
+            }
+            break;
+        }
+
+        /* clang-format off */
+        data_t *data = data_make(
+                "model",            "",     DATA_STRING,    "Honeywell-Remote",
+                "command",          "",     DATA_STRING,    command_string,
+                "dimmable",         "",     DATA_INT   ,    dimmable,
+                "mic",              "",     DATA_STRING,    "PARITY",
```

Comment:
MIC stands for message integrity check. Right now we have checksum, crc and parity. From the description I see neither of those. The point of the tag is how much can we trust the message. I always wanted to add repeats as a mic, and here it would fit well with some changes (bitbuffer_find_repeated_row). Then we have the question how many repeats should be required for a mic tag and here I think 3-5 should a good range depending on the length of the message.